inception-project/inception
# #1013 - Expand use of SPARQLQueryBuilder to parent/child-axis

**What's in the PR**
- Introduced new SPARQLQueryBuilderGenericTest which runs tests not requiring explicit test data on all KBs we have profiles for (except some which are known to fail)
- Added support for property paths in SPARQL queries
- Factored our VALUES queries into a separate class
- Tiered builder pattern for SPARQLQueryBuilder
- Fixed various problems with Virtuoso by massaging/improving queries
- Added support for child/descendant/parent/ancestor queries
- Removed corresponding queries from SPARQLQueryStore
- Use SPARQLQueryBuilder for child/descendant/parent/ancestor queries in KnowledgeBaseServiceImpl
- Updates test data in KnowledgeBaseServiceRemoteTest to actually do something sensible on DBPedia and Yago
- Removed tests from KnowledgeBaseServiceRemoteTest which were superseded by tests in the SPARQLQueryBuilderGenericTest or SPARQLQueryBuilderTest
- Fixes #1008 - KnowledgeBaseServiceRemoteTest.thatChildConceptsCanBeRetrieved[KB = Wikidata (official/direct mapping)] fails 
- Fixes #962 - Add graph IRI setting to KB settings (adds setting to the KB, to the profiles, and to the UI)
- Fixes #1011 - Restore owl:intersectionOf support
- Fixed javadoc complaint
- Fixed bug that root concepts accumulate when selecting different KB profiles in the wizard
- Disable server-side reduce by default because we cannot guarantee the right language to be returned
- Added JavaDoc to various methods
- Enabled FTS for DBPedia profile
- Switch from deprecated Repository.initialize() to modern Repository.init()
- Updated documentation
- Depend on release version of INCEpTION RDF4J fork
- Sort results from SPARQLQueryBuilder case-insensitive for a better user experience
- Avoid displaying duplicate statements

**How to test manually**
* Import some of the pre-configured KBs and check if they work (mind, BabelNet is having trouble, but I don't think it is our fault).

**Automatic testing**

**Documentation**
